### PR TITLE
dsp/window: add Tukey, clean up tests, simplify API and fix scaling infinities by zero

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

### CODE REVIEW

File: dsp/window/window_complex.go [hunk 1]
```diff
@@ -79,8 +81,9 @@ func LanczosComplex(seq []complex128) []complex128 {
 // Spectral leakage parameters: ΔF_0 = 4, ΔF_0.5 = 1.33, K = 2, ɣ_max = -26.5, β = -6.
 func TriangularComplex(seq []complex128) []complex128 {
 	a := float64(len(seq)-1) / 2
-	for i := range seq {
-		seq[i] *= complex(1-math.Abs(float64(i)/a-1), 0)
+	for i, v := range seq {
+		w := 1 - math.Abs(float64(i)/a-1)
```
Comment (romanwerpachowski) [inline dsp/window/window_complex.go:85]: The window is symmetric, isn't there some optimisation opportunity? we could avoid calling math.Abs.

File: dsp/window/window_parametric.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	if len(v) != len(seq) {
+		panic("window: length mismatch")
+	}
+	for i, w := range v {
+		seq[i] *= w
+	}
+	return seq
+}
+
+// Transform applies the weights in the receiver to seq in place, returning the
```
Comment (vladimir-ch) [inline dsp/window/window_parametric.go:150]: TransformComplex

File: dsp/window/window_complex.go (hunk 1 above)
Comment (kortschak) [reply dsp/window/window_complex.go:85]: Yeah. They're all symmetric. Is it OK if we get to that later (we already have a perfect optimisation by using the `Values` type)?

Comment (romanwerpachowski) [reply dsp/window/window_complex.go:85]: Sure.

File: dsp/window/window_parametric.go (hunk 2 above)
Comment (kortschak) [reply dsp/window/window_parametric.go]: Done

File: dsp/window/window_parametric.go [hunk 3]
```diff
@@ -147,31 +147,19 @@ func (v Values) Transform(seq []float64) []float64 {
...
-	for i := range v {
-		v[i] = 1
-	}
-	return window(v)
-}
-
-// Transform applies the weights in the receiver to seq in place, returning the
-// result. If v is nil, Transform is a no-op, otherwise the length of v must
+// TransformComplex applies the weights in the receiver to seq in place, returning
+// the result. If v is nil, Transform is a no-op, otherwise the length of v must
```
Comment (romanwerpachowski) [inline dsp/window/window_parametric.go:151]: I don't understand this change, TBH. weighting floats with complex numbers is not a replacement for weighting complex numbers with complex numbers?

File: dsp/window/window_parametric.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+		panic("window: length mismatch")
+	}
+	for i, w := range v {
+		seq[i] *= w
+	}
+	return seq
+}
+
+// TransformComplex applies the weights in the receiver to seq in place, returning
+// the result. If v is nil, Transform is a no-op, otherwise the length of v must
```
Comment (vladimir-ch) [inline dsp/window/window_parametric.go:151]: TransformComplex also here. Sorry I didn't see it before.

File: dsp/window/window_parametric.go (hunk 3 above)
Comment (kortschak) [reply dsp/window/window_parametric.go:151]: You're absolutely right; the weighting (I'm sort of reluctant to call it weighting) complex numbers with complex numbers should not have been here. The point of the package is to provide scalar weighting of inputs for Fourier transforms. The approach that was used before was a lazy way to apply scalar weightings (my blame).

File: dsp/window/window_parametric.go (hunk 4 above)
Comment (kortschak) [reply dsp/window/window_parametric.go]: ... or I could use some initiative.

File: dsp/window/window_parametric.go (hunk 3 above)
Comment (romanwerpachowski) [reply dsp/window/window_parametric.go]: I get it now. Thanks.

**Review by romanwerpachowski — APPROVED** [review]

**Review by vladimir-ch — APPROVED** [review]
